### ISSUE TITLE
Prefer if over match with single-case

### ISSUE THREAD

**Issue by ZhdanovAM72** [wemake-services/wemake-python-styleguide#3528]
# Rule request

## Thesis
We should support simplifying `match` statements with only one case and no wildcard:

Example:
```python
match x:
    case 1:
        do_something()
```

Can be rewritten as:
```python
if x == 1:
    do_something()
```

## Reasoning
Even without an else branch, a single-case `match` is unnecessarily complex compared to a simple `if`. The intent is clear, and using `if` reduces nesting and cognitive load.

This is especially common in validation or dispatch logic where only one condition matters.

When should this be allowed?

Only when:
1. There is exactly one case
2. The pattern is simple (literal, constant, enum, etc.)
3. No guard (`if` ...) is used

**Comment by sobolevn** [comment]
No, we don't need any configuration here. Any `match` with just one `case` can be converted to `if`, see https://github.com/wemake-services/wemake-python-styleguide/commit/3246459d13023a7954b29654a8b2727f1bb57976

You can reuse its logic and violation. Only need new tests and new condition.
